clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java

```diff
@@ -637,7 +637,15 @@ private ConsumerRecords<K, V> poll(final Timer timer, final boolean includeMetad
                                 + "since the consumer's position has advanced for at least one topic partition");
                     }
 
-                    return this.interceptors.onConsume(new ConsumerRecords<>(fetch.records()));
+                    Map<TopicPartition, OffsetAndMetadata> nextOffsetAndMetadata = new HashMap<>();
+                    Map<TopicPartition, Long> nextFetchOffsets = fetch.nextFetchOffsets();
```

Comment:
If both nextFetchOffsets and lastEpochs are coming from the fetch, could we just create `OffsetAndMetadata` directly inside a map `fetch.nextOffsetAndMetadata`? We'd avoid dealing with multiple maps this way it seems.